src/main/java/Database.java

```diff
@@ -0,0 +1,75 @@
+import java.util.ArrayList;
+
+public class Database {
+    public static ArrayList<Task> taskList = new ArrayList<>();
+
+    public static void markDone(String arg) {
+        try{
+            int index = Integer.parseInt(arg);
+            taskList.get(index - 1).markAsDone();
+            markDoneResponse(index);
+        } catch(NumberFormatException e) {
+            System.out.println("The command is not recognised.");
+        } catch (IndexOutOfBoundsException e){
+            System.out.println("The index is out of range. This action is aborted.");
+        }
+    }
+
+    public static void listAll(){
+        int i = 1;
+        for(Task task: taskList){
+
+            System.out.println(i + ". " + task);
+            i++;
+        }
+    }
+
+    public static void addToDo(String args){
+        Task todo = new ToDo(args);
+        taskList.add(todo);
+        addedToListResponse(todo);
+    }
+
+    public static void addDeadline(String args){
+        String[] parts = args.split(Constants.BY_PARSER);
+        Task ddl;
+        try{
+            ddl = new Deadline(parts[0], parts[1]);
+        } catch (ArrayIndexOutOfBoundsException e){
+            System.out.println("Not provided sufficient arguments!");
+            return;
+        }
+        taskList.add(ddl);
```

Comment:
I like how you have put them in different functions for different types of inputs by refactoring the code. Makes the code more readable.